tests/test_torch_wl_kernel.py

```diff
@@ -0,0 +1,111 @@
+from __future__ import annotations
+
+import networkx as nx
+import pytest
+import torch
+from grakel import WeisfeilerLehman, graph_from_networkx
+from grakel_replace.torch_wl_kernel import TorchWLKernel, GraphDataset
+
+
+@pytest.mark.parametrize("n_iter", [1, 2, 3, 4, 5, 6, 7])
+@pytest.mark.parametrize("normalize", [True, False])
+def test_wl_kernel_against_grakel(n_iter, normalize):
+    """Test the custom WL kernel against Grakel's implementation.
+
+    Args:
+        n_iter: Number of iterations for the WL kernel.
+        normalize: Whether to normalize the kernel matrix.
+    """
+    # Create example graphs for testing
+    G1 = nx.Graph()
+    G1.add_edges_from([(0, 1), (1, 2), (1, 3), (1, 4), (2, 3)])
+    for node in G1.nodes():
+        G1.nodes[node]["label"] = str(node)
+
+    G2 = nx.Graph()
+    G2.add_edges_from([(0, 1), (1, 2), (2, 3), (3, 4)])
+    for node in G2.nodes():
+        G2.nodes[node]["label"] = str(node)
+
+    G3 = nx.Graph()
+    G3.add_edges_from([(0, 1), (1, 3), (3, 2)])
+    for node in G3.nodes():
+        G3.nodes[node]["label"] = str(node)
+
+    graphs = GraphDataset.from_networkx([G1, G2, G3])
+
+    # Initialize and compute kernel matrix using custom WLKernel
+    wl_kernel = TorchWLKernel(n_iter=n_iter, normalize=normalize)
+    torch_kernel_matrix = wl_kernel(graphs).cpu().detach().numpy()
+
+    # Convert to Grakel-compatible format
+    grakel_graphs = graph_from_networkx([G1, G2, G3], node_labels_tag="label")
+
+    # Initialize and compute kernel matrix using Grakel's Weisfeiler-Lehman kernel
+    grakel_wl = WeisfeilerLehman(n_iter=n_iter, normalize=normalize)
+    grakel_kernel_matrix = grakel_wl.fit_transform(grakel_graphs)
+
+    # Assert that the kernel matrices are similar within a reasonable tolerance
+    assert torch.allclose(
+        torch.tensor(torch_kernel_matrix, dtype=torch.float64),
+        torch.tensor(grakel_kernel_matrix, dtype=torch.float64),
+        atol=1e-5
```

Comment:
Why `1e-5`, seems like we should be able to get even better precision than that? Unlikely to really matter but just curious?